server/src/main/java/io/unitycatalog/server/service/TemporaryModelVersionCredentialsService.java

```diff
@@ -0,0 +1,59 @@
+import io.unitycatalog.server.exception.ErrorCode;
+import io.unitycatalog.server.exception.GlobalExceptionHandler;
+import io.unitycatalog.server.model.*;
+import io.unitycatalog.server.persist.ModelRepository;
+import io.unitycatalog.server.persist.utils.RepositoryUtils;
+import io.unitycatalog.server.service.credential.CredentialOperations;
+import io.unitycatalog.server.service.credential.CredentialContext;
+import io.unitycatalog.server.persist.ModelRepository;
+
+import java.util.Collections;
+import java.util.Set;
+
+import static io.unitycatalog.server.service.credential.CredentialContext.Privilege.SELECT;
+import static io.unitycatalog.server.service.credential.CredentialContext.Privilege.UPDATE;
+
+@ExceptionHandler(GlobalExceptionHandler.class)
+public class TemporaryModelVersionCredentialsService {
+
+    private static final ModelRepository MODEL_REPOSITORY = ModelRepository.getInstance();
+
+    private final CredentialOperations credentialOps;
+
+    public TemporaryModelVersionCredentialsService(CredentialOperations credentialOps) {
+        this.credentialOps = credentialOps;
+    }
+
+    @Post("")
+    public HttpResponse generateTemporaryModelVersionCredential(
+            GenerateTemporaryModelVersionCredential generateTemporaryModelVersionCredential) {
+
+        long modelVersion = generateTemporaryModelVersionCredential.getVersion();
+        String catalogName = generateTemporaryModelVersionCredential.getCatalogName();
+        String schemaName = generateTemporaryModelVersionCredential.getSchemaName();
+        String modelName = generateTemporaryModelVersionCredential.getModelName();
+        String fullName = RepositoryUtils.getAssetFullName(catalogName, schemaName, modelName);
+
+        ModelVersionInfo modelVersionInfo = MODEL_REPOSITORY.getModelVersion(fullName, modelVersion);
+        ModelVersionOperation requestedOperation = generateTemporaryModelVersionCredential.getOperation();
+        // Must enforce that write credentials are not passed back if the model has been finalized
+        if (modelVersionInfo.getStatus() != ModelVersionStatus.PENDING_REGISTRATION && requestedOperation == ModelVersionOperation.READ_WRITE_MODEL_VERSION) {
+            throw new BaseException(ErrorCode.INVALID_ARGUMENT, "Cannot request read/write credentials on a finalized model version: " + fullName + "/" + modelVersion);
+        }
+        return HttpResponse.ofJson(credentialOps.vendCredentialForModelVersion(modelVersionInfo, modelVersionOperationToPrivileges(generateTemporaryModelVersionCredential.getOperation())));
+    }
+
+    private Set<CredentialContext.Privilege> modelVersionOperationToPrivileges(ModelVersionOperation modelVersionOperation) {
+        return switch (modelVersionOperation) {
+            case READ_MODEL_VERSION -> Set.of(SELECT);
+            case READ_WRITE_MODEL_VERSION -> Set.of(SELECT, UPDATE);
+            case UNKNOWN_MODEL_VERSION_OPERATION -> Collections.emptySet();
```

Comment:
shouldn't we throw error in this case ?

Reply:
cc @alexreid-db on this as we would need to fix it everywhere if the answer is yes.

Reply:
I'm confused when a client would ever send this in the request, but sure we can throw error here. Probably should be `default -> throw new ...`

Reply:
What if I a client is deliberately trying to be malicious. I was just worried about the case when`UnknownOperation` can by mistake translate to credentials which have the most broad access.

Reply:
I'm more commenting on why this value is even a part of api / spec if we are just going to always throw error if sent? (just don't include it as an enum value). If a client sends an value that isn't in the spec, it won't deserialize and the server will throw an exception on its own...

Reply:
The other thing to mention is that once the access control story makes it into the codebase, the credential vending would always check what privileges the subject/user/principal and check against what is being asked for and never elevate privileges.

Potentially this unknown value would be more used for just having the server send back credentials for privileges that the subject/user/principal has (i.e. the client doesn't tell the server what operations it wants and the server just gives what they have access to), but no idea if that is the actual purpose. 

How the iceberg api works is that it always gives you credentials with privileges you have for a table (instead of the client asking for certain privileges), which is a little more straightforward/simpler.

Reply:
but all that being said, I'm fine with just throwing an error here...